CDCgov/prime-reportstream
# Adjust sender transforms to prevent exceptions

This PR aims to prevent exceptions like the following from happening in staging/prod due to unexpected issues in the messages being sent.

<details><summary>Exception log</summary>

> Exception while executing function: Functions.convert-fhir Result: Failure
> Exception: FHIRException: Attempt to add child with unknown name value
> Stack: java.lang.reflect.InvocationTargetException
> 	at jdk.internal.reflect.GeneratedMethodAccessor380.invoke(Unknown Source)
> 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
> 	at com.microsoft.azure.functions.worker.broker.JavaMethodInvokeInfo.invoke(JavaMethodInvokeInfo.java:22)
> 	at com.microsoft.azure.functions.worker.broker.EnhancedJavaMethodExecutorImpl.execute(EnhancedJavaMethodExecutorImpl.java:22)
> 	at com.microsoft.azure.functions.worker.chain.FunctionExecutionMiddleware.invoke(FunctionExecutionMiddleware.java:19)
> 	at com.microsoft.azure.functions.worker.chain.InvocationChain.doNext(InvocationChain.java:21)
> 	at com.microsoft.azure.functions.worker.broker.JavaFunctionBroker.invokeMethod(JavaFunctionBroker.java:125)
> 	at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:34)
> 	at com.microsoft.azure.functions.worker.handler.InvocationRequestHandler.execute(InvocationRequestHandler.java:10)
> 	at com.microsoft.azure.functions.worker.handler.MessageHandler.handle(MessageHandler.java:44)
> 	at com.microsoft.azure.functions.worker.JavaWorkerClient$StreamingMessagePeer.lambda$onNext$0(JavaWorkerClient.java:94)
> 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
> 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
> 	at java.base/java.lang.Thread.run(Thread.java:833)
> Caused by: org.hl7.fhir.exceptions.FHIRException: Attempt to add child with unknown name value
> 	at org.hl7.fhir.r4.model.Base.addChild(Base.java:177)
> 	at org.hl7.fhir.r4.model.Element.addChild(Element.java:289)
> 	at org.hl7.fhir.r4.model.Extension.addChild(Extension.java:478)
> 	at gov.cdc.prime.router.fhirengine.translation.hl7.FhirTransformer.setBundleProperty$prime_router(FhirTransformer.kt:206)
> 	at gov.cdc.prime.router.fhirengine.translation.hl7.FhirTransformer.transformBasedOnElement$prime_router(FhirTransformer.kt:129)
> 	at gov.cdc.prime.router.fhirengine.translation.hl7.FhirTransformer.transformWithSchema(FhirTransformer.kt:70)
> 	at gov.cdc.prime.router.fhirengine.translation.hl7.FhirTransformer.transformWithSchema$default(FhirTransformer.kt:56)
> 	at gov.cdc.prime.router.fhirengine.translation.hl7.FhirTransformer.transform(FhirTransformer.kt:48)
> 	at gov.cdc.prime.router.fhirengine.engine.FHIRConverter.doWork(FHIRConverter.kt:69)
> 	at gov.cdc.prime.router.fhirengine.engine.FHIREngine.run(FHIREngine.kt:173)
> 	at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions$runFhirEngine$newMessages$1.invoke(FHIRFunctions.kt:151)
> 	at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions$runFhirEngine$newMessages$1.invoke(FHIRFunctions.kt:150)
> 	at gov.cdc.prime.router.azure.DatabaseAccess.transactReturning$lambda$1(DatabaseAccess.kt:106)
> 	at org.jooq.impl.DefaultDSLContext.lambda$transactionResult0$3(DefaultDSLContext.java:531)
> 	at org.jooq.impl.Tools$3$1.block(Tools.java:6068)
> 	at java.base/java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3463)
> 	at java.base/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3434)
> 	at org.jooq.impl.Tools$3.get(Tools.java:6065)
> 	at org.jooq.impl.DefaultDSLContext.transactionResult0(DefaultDSLContext.java:579)
> 	at org.jooq.impl.DefaultDSLContext.transactionResult(DefaultDSLContext.java:503)
> 	at gov.cdc.prime.router.azure.DatabaseAccess.transactReturning(DatabaseAccess.kt:106)
> 	at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions.runFhirEngine(FHIRFunctions.kt:150)
> 	at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions.doConvert$prime_router(FHIRFunctions.kt:60)
> 	at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions.doConvert$prime_router$default(FHIRFunctions.kt:54)
> 	at gov.cdc.prime.router.fhirengine.azure.FHIRFunctions.convert(FHIRFunctions.kt:45)
> 	... 17 more

</details>

To prevent the issue, anywhere we could be adding `value` to an extension when processing the `bundleProperty` I moved the `value` piece into the `resource` instead. This means that if the value doesn't already exist, that element will be a no-op, and the exception won't occur. In each of the changes cases, that should be okay, since we probably don't need to run the transform if the value is missing.

Test Steps:
1. Run the latest integration/translation tests.
2. See that no exceptions are thrown and that the tests pass.

## Checklist

### Testing
